kernel/src/process/posix_thread/thread_local.rs

```diff
@@ -83,6 +83,10 @@ impl ThreadLocal {
     pub fn sig_stack(&self) -> &RefCell<Option<SigStack>> {
         &self.sig_stack
     }
+
+    pub fn set_file_table(&self, new_table: Option<RwArc<FileTable>>) {
+        *self.file_table.borrow_mut() = new_table;
```

Comment:
The new `set_file_table` method does not match well with the existing  `borrow_file_table` and `borrow_file_table_mut` methods (they are not `get_file_table`, `remove_file_table`, etc.`). A more consistent approach is to add a `replace` method to `FileTableRefMut`.

```rust
impl FileTableRefMut<'_> {
    pub fn replace(&mut self, new_table: Option<RwArc<FileTable>>) -> Option<RwArc<FileTable>> { ... }
}
```